### PR TITLE
React Capstone: 

### PR DESCRIPTION
**Pull request by KanzaTahreem**
For the React Capstone project, I have selected an API that fetches data from [CoinGecko API](https://www.coingecko.com/en/api/documentation).
- As per requirements, The [CoinGecko API](https://www.coingecko.com/en/api/documentation) provides numeric data about cryptocurrencies.
- The app is built around two pages:
  - Home page
    - When the page loads, the web app retrieves data from the selected API and display a list of item on the screen.
    - The consists of a list of categories that can be filtered out by category name.
    - Along with the category name, the app displays numeric values per category that come from the API.
    - When the user clicks (or taps) on a category item, the application navigates to the details page.
  
  - Details page
    - In the details page, the web app retrieves data from the API to show detailed data in the category.
    - When the user clicks on the _Back_ button (<), the user navigates to the home page.
 - The app is tested through [React Testing Library](https://testing-library.com/docs/react-testing-library/intro/) and jest snapshots.
 
 Here's the [video description](https://loom.com/share/b75aeada007d478bac3e2bf311105a51) of the project
And Here is the [Live Link](https://kanzatahreem.github.io/Crypto-Analysis/)

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { Routes, Route } from 'react-router-dom';
+import { useDispatch } from 'react-redux';
+import { useEffect } from 'react';
+import Navbar from './components/Navbar';
+// import Header from './components/Header';
```
Comment (rloterh) [inline src/App.js]: - [ ] OPTIONAL
There are commented code blocks found in your code on the repository. Kindly endeavor to remove **all** commented code from your code before pushing it to the repository so as to keep your code well organized.

File: src/Pages/CoinsList.jsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import React from 'react';
+import millify from 'millify';
+import { useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+import CoinsData from '../components/CoinsData';
+import Header from '../components/Header';
+import '../styles/CoinsList.css';
+
```
Comment (rloterh) [inline src/Pages/CoinsList.jsx:8]: The styling of the list on your Homepage does not conform to the alternating dual-color pattern of the provided [design template](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)), as well as the column layout of cells. Kindly note we as much as possible want your design to match the provided [template](https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)). Therefore kindly ensure that:
- [ ] Your design **fits** perfectly on mobile screens
- [ ] The cards/ cells on the _Home page_ are **two columns** on mobile.
- [ ] The cells/ card of your list follow the **alternating dual-color pattern**. Each cell of the list item should have surrounding cells having **different (alternating) background color(s)**.

**Your design:**        |  **The original design:**
:-------------------------:|:-------------------------:
Home Page | Home Page

File: README.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+```
+  npm run deploy
+```
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- AUTHORS -->
+
+## 👥 Authors <a name="authors"></a>
```
Comment (rloterh) [inline README.md]: - [ ] OPTIONAL
- I think you should change the header title _Authors_ to _Author_ in your ReadMe, as this project is worked on by a sole contributor. You can optionally also change it to _Author(s)_ to create room for future contributors.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Kanza, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦   

## Highlights 👍  
- Nice presentation of this project
- Good job acknowledging Nelson Sakwa on Behance for the [Original design idea](https://www.behance.net/sakwadesignstudio).
- API fetch properly implemented
- State management with Redux properly implemented
- Navigation working as intended
- Filter works great
- Good unit test coverage
- Great ReadMe

## Required Changes ♻️
_Kindly check the inline comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

_Kindly check the inline comments marked **[OPTIONAL]** under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by algerina — APPROVED** [review]
Hi @KanzaTahreem , 
Good job so far!

May I highlight the following:

- The repo looks professional.:heavy_check_mark:
- There are no linter errors:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:
- Your project looks awesome:heavy_check_mark:

## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:
Your project is complete!  It's time to merge it :shipit:
Great work on implementing the changes requested by the previous TSEs


 
 ## Optional suggestions
 
 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Pull request merged**